benchmark/src/test/java/ai/timefold/solver/benchmark/impl/aggregator/BenchmarkAggregatorTest.java

```diff
@@ -0,0 +1,258 @@
+        assertThat(htmlReportFile.getName()).endsWith(".html");
+    }
+
+    @Test
+    void getAvailableBenchmarkDirectories_nonExistentDirectory() {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        config.setBenchmarkDirectory(new File("non/existent/directory"));
+
+        List<String> directories = benchmarkAggregator.getAvailableBenchmarkDirectories(config);
+
+        assertThat(directories).isEmpty();
+    }
+
+    @Test
+    void getAvailableBenchmarkDirectories_emptyDirectory(@TempDir Path tempDir) {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        config.setBenchmarkDirectory(tempDir.toFile());
+
+        List<String> directories = benchmarkAggregator.getAvailableBenchmarkDirectories(config);
+
+        assertThat(directories).isEmpty();
+    }
+
+    @Test
+    void getAvailableBenchmarkDirectories_withValidDirectories() throws IOException {
+        PlannerBenchmarkConfig config = PlannerBenchmarkConfig
+                .createFromXmlResource("ai/timefold/solver/benchmark/api/testdataBenchmarkConfigWithReport.xml");
+
+        List<String> directories = benchmarkAggregator.getAvailableBenchmarkDirectories(config);
+
+        assertThat(directories)
+                .hasSize(4)
+                .containsExactlyInAnyOrder("2025-09-05_141529", "2025-09-05_141529_1", "2025-09-05_141530",
+                        "2025-09-05_141530_1");
+    }
+
+    @Test
+    void aggregateAndShowReportInBrowser_nonExistentDirectory() {
+        PlannerBenchmarkConfig config = PlannerBenchmarkConfig
+                .createFromXmlResource("ai/timefold/solver/benchmark/api/testdataBenchmarkConfigWithReport.xml");
+        File nonExistentDir = new File("non/existent/directory");
+        config.setBenchmarkDirectory(nonExistentDir);
+
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> benchmarkAggregator.aggregateAndShowReportInBrowser(config))
+                .withMessageContaining("Benchmark directory does not exist");
+    }
+
+    @Test
+    void aggregateAndShowReportInBrowser_emptyDirectory(@TempDir Path tempDir) {
+        PlannerBenchmarkConfig config = createValidBenchmarkConfig(tempDir);
+
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> benchmarkAggregator.aggregateAndShowReportInBrowser(config))
+                .withMessageContaining("No benchmark results found in directory");
+    }
+
+    @Test
+    void aggregateAndShowReportInBrowser_withValidResults() {
+        PlannerBenchmarkConfig config = PlannerBenchmarkConfig
+                .createFromXmlResource("ai/timefold/solver/benchmark/api/testdataBenchmarkConfigWithReport.xml");
+
+        File htmlReportFile = benchmarkAggregator.aggregateAndShowReportInBrowser(config);
+
+        assertThat(htmlReportFile).isNotNull();
+        assertThat(htmlReportFile.exists()).isTrue();
+        assertThat(htmlReportFile.getName()).endsWith(".html");
```

Comment:
This is another test which requires copies of the benchmark reports in resources. We want to check that the benchmarker understand the reports _today_, not some reports from long past. 

(Even though we do not intend to break those reports, we do not commit to backwards compatibility there.)

IMO this is exactly the test that requires running several very short benchmarks (milliseconds) and aggregating them on the fly. An end-to-end test, without relying on report copies in resources.